tsconfig.es5.json

```diff
@@ -1,6 +1,7 @@
 {
   "extends": "./tsconfig.base.json",
   "compilerOptions": {
-    "target": "es5"
+    "target": "es5",
+    "downlevelIteration": true
```

Comment:
typescript is complaining on https://github.com/open-telemetry/opentelemetry-js/blob/main/experimental/packages/opentelemetry-sdk-metrics-base/src/state/MetricStorageRegistry.ts#L31 that ES6 iterators need to be transpiled to work on ES5.

Reply:
Submitted #2823 to fix this.